### ISSUE TITLE
attribute JSON - determiner

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#9224]
Working on https://github.com/ArctosDB/dev/issues/406, attributes (record at the moment, but this will apply to all types) don't have sufficient information in the cache. They're currently:

```json
[
  {
    "attribute_date": "2004-02-26",
    "attribute_type": "sex",
    "attribute_units": null,
    "attribute_value": "female",
    "attribute_method": null,
    "attribute_remark": null,
    "attribute_determiner": "University of Alaska Museum"
  }
]
```

I've considered a simple/"flat" approach:

```json
[
  {
    "attribute_date": "2004-02-26",
    "attribute_type": "sex",
    "attribute_units": null,
    "attribute_value": "female",
    "attribute_method": null,
    "attribute_remark": null,
    "attribute_determiner": "University of Alaska Museum",
    "attribute_determinerid": "https://arctos.database.museum/agent/3691"
  }
]
```

or a more normalized view:

```json
[
  {
    "attribute_date": "2004-02-26",
    "attribute_type": "sex",
    "attribute_units": null,
    "attribute_value": "female",
    "attribute_method": null,
    "attribute_remark": null,
    "attribute_determiner": {
      "agentID": "https://arctos.database.museum/agent/3691",
      "preferred_name": "University of Alaska Museum"
    }
  }
]
```

I can't think of any real technical concerns, does anyone have any preferences?? Both seem "simpler" or "friendlier" in various ways, maybe the second is 'more correct' so I suppose I'll go with it if there's no reason not to??

Key name suggestions also appreciated!

HELP!

**Comment by Jegelewicz** [comment]
I vote for both

**Comment by dustymc** [comment]
>both

I don't understand, we can only have one format?

**Comment by dustymc** [comment]
Possibly there's a compelling reason to normalize in Identifiers, where I need to add ***two*** agents eg

```json
[
  {
    "remarks": null,
    "issued_by": "Aaron M. Bauer",
    "identifier": "AMB 53",
    "assigned_by": "unknown",
    "assigned_date": "2022-04-27",
    "identifier_type": "collector number"
  }
]
```

I'll edit the title of this Issue - whatever we do should be done as broadly as possible, and that clearly extends beyond attributes.

**Comment by dustymc** [comment]
I'll also update identifications with whatever we come up with here. It currently has the data I need for determiner, but not following current identifier naming conventions:

```
"identification_agents": [
      {
        "agent_name": "W. v. Koenigswald",
        "agent_identifier": "https://arctos.database.museum/agent/7769",
        "identifier_order": 1
      },
      {
        "agent_name": "Elizabeth R. Dumont",
        "agent_identifier": "https://arctos.database.museum/agent/21311664",
        "identifier_order": 2
      },
....

```

and attributes are more "flat" - and I think probably confusing, it's not entirely clear why the agent is there, wrapping the first two pairs up in a 'determined by' label would, I think, help:

```
"identification_attributes": [
      {
        "agentid": "https://arctos.database.museum/agent/21256093",
        "agent_name": "Candice Johnson",
        "attribute_type": "identification confidence",
        "attribute_units": null,
        "attribute_value": "medium",
        "determined_date": "2025-03-11",
        "attribute_remark": null,
        "determination_method": null
      },

```

**Comment by dustymc** [comment]
I think the flat option just isn't going to scale, consistency is always good, and it's not dificult to follow our identifier convention (at least in JSONB?), so I'm going to propose:

```json
[
  {
    "agent_role_or_whatever": {
      "agentID": "https://arctos.database.museum/agent/3691",
      "agent_name": "University of Alaska Museum"
    }
  }
]


```

for use anywhere agents are found in JSON. Here's a full record attribute:


```json
[
  {
    "attribute_date": "2004-02-26",
    "attribute_type": "sex",
    "attribute_units": null,
    "attribute_value": "female",
    "attribute_method": null,
    "attribute_remark": null,
    "attribute_determiner": {
      "agentID": "https://arctos.database.museum/agent/3691",
      "agent_name": "University of Alaska Museum"
    }
  }
]
```

and one without agent info, in case that's useful:

```json
[
  {
    "attribute_date": null,
    "attribute_type": "individual count",
    "attribute_units": "individuals",
    "attribute_value": "1",
    "attribute_method": "quantity_total from morpho_ids in ParaSite database",
    "attribute_remark": null,
    "attribute_determiner": {
      "agentID": null,
      "agent_name": null
    }
  },
  {
    "attribute_date": null,
    "attribute_type": "verbatim host ID",
    "attribute_units": null,
    "attribute_value": "Passerella iliaca",
    "attribute_method": "Passerella iliaca",
    "attribute_remark": null,
    "attribute_determiner": {
      "agentID": null,
      "agent_name": null
    }
  }
]
```

Please let me know ASAP if I'm missing some fatal flaw, possible simplification, whatever....
